chain/consensus/mir/mine.go

```diff
@@ -1,76 +1,159 @@
 package mir
 
 import (
+	"bytes"
 	"context"
 	"fmt"
 	"time"
 
+	"go.uber.org/zap/buffer"
+
 	"github.com/filecoin-project/go-address"
 	lapi "github.com/filecoin-project/lotus/api"
 	"github.com/filecoin-project/lotus/api/v1api"
 	"github.com/filecoin-project/lotus/chain/consensus/platform/logging"
 	"github.com/filecoin-project/lotus/chain/types"
 	ltypes "github.com/filecoin-project/lotus/chain/types"
+	mirproto "github.com/filecoin-project/mir/pkg/pb/requestpb"
 )
 
-// Mine implements "block mining" using Mir framework.
+// Mine implements "block mining" using the Mir framework.
 //
-// Mine implements the following algorithm:
+// Mine implements the following main algorithm:
 // 1. Retrieve messages and cross-messages from the mempool.
-//    Note, that messages can be added into mempool via the libp2p mechanism and the CLI.
+//    Note, that messages can be added into mempool via the libp2p and CLI mechanism.
 // 2. Send messages and cross messages to the Mir node through the request pool implementing FIFO.
 // 3. Receive ordered messages from the Mir node and parse them.
-// 4. Create the next Filecoin block. Note, only a leader Eudico node, chosen by round-robin election, creates a block.
+// 4. Create the next Filecoin block. Note, only the leader Eudico node, chosen by round-robin election,
```

Comment:
We should definitely change this and make a Reward interface that allows, according to the consensus algorithm, to perform a different reward distribution. I guess in Mir's case we should distribute the rewards between all the validators.

Reply:
Added - https://github.com/filecoin-project/eudico/issues/239